kotlin/samples/containers/android/src/main/java/com/squareup/sample/container/BackButtonScreen.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sample.container
+
+import com.squareup.workflow.ui.BuilderBinding
+import com.squareup.workflow.ui.ViewBinding
+import com.squareup.workflow.ui.ViewRegistry
+import com.squareup.workflow.ui.backPressedHandler
+import com.squareup.workflow.ui.bindShowRendering
+import com.squareup.workflow.ui.getShowRendering
+
+/**
+ * Adds optional back button handling to a [wrapped] rendering, possibly overriding its own.
+ *
+ * @param [override] If true, and [onBackPressed] is set, any back button handler
+ * set by the [wrapped] rendering is overridden. If false, a handler set by
+ * [wrapped] takes precedence. Defaults to `false`.
+ *
+ * @param [onBackPressed] The function to fire when the device back button
+ * is pressed, or null to set no handler. Defaults to `null`.
+ */
+data class BackButtonScreen<W : Any>(
+  val wrapped: W,
+  val override: Boolean = false,
+  val onBackPressed: (() -> Unit)? = null
+) {
+  companion object Binding : ViewBinding<BackButtonScreen<*>>
```

Comment:
This binding is reasonably complex, it should have unit tests.

Reply:
Tricky, this is view code. I'm more inclined to add an espresso test, and beef up the app a tiny bit to cover more use cases.